### PR TITLE
Add support for classes to isometric objects.

### LINKED ISSUE #216: Support CSS Classes?
Instead of directly specifying properties like `fillColor`, I'd like to be able to use classes.

Why? Primarily because I want to support dark mode. It's much easier to apply a class to an element and allow it to change automatically based on `@media (prefers-color-scheme: dark) { ... }` rather than handling the event `window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', event => { ... })`.

### PR DESCRIPTION
**Pull request by joelburget**
Fixes #216

### CODE REVIEW

**Comment by elchininet** [comment]
Hi @joelburget,
Thanks for contributing 🙂
Some things that I have observed in a quick look, please, fix them and I'll review the code with more attention:

1. The `className` should not have an empty string as a default value, this parameter should be added only if it has a value. The snapshots that are not using that option should not have an empty `class` parameter.
2. There should be tests cases for this new option
3. This new option should be added to the documentation to each class
 
Regards and thanks again for contributing to the project.

**Comment by joelburget** [comment]
I think I fulfilled all of your requests, please let me know how it looks.

**Comment by elchininet** [comment]
> I think I fulfilled all of your requests, please let me know how it looks.

@joelburget, looks good at first sight. I'll take a deeper look when I have some time. 👍

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,11395 @@
+{
```
Comment (elchininet) [inline package-lock.json:1]: Remove this one

File: src/@types/type-modules/svg.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { StringOrNumber } from './generic';
 export type StrokeLinecap = keyof typeof LineCap;
 export type StrokeLinejoin = keyof typeof LineJoin;
 
-export type SVGProperties = 'fillColor' | 'fillOpacity' | 'strokeColor' | 'strokeOpacity' | 'strokeWidth';
+export type SVGProperties = 'fillColor' | 'fillOpacity' | 'strokeColor' | 'strokeOpacity' | 'strokeWidth' | 'className';
```
Comment (elchininet) [inline src/@types/type-modules/svg.ts:7]: The name of the type is confusing. The `className` property should not be here. This type is used only for properties that are animatable.

File: src/@utils/svg.ts [hunk 3]
```diff
@@ -162,7 +162,8 @@ export const isSVGProperty = (property: string): property is SVGProperties => {
         'fillOpacity',
         'strokeColor',
         'strokeOpacity',
-        'strokeWidth'
+        'strokeWidth',
+        'className'
```
Comment (elchininet) [inline src/@utils/svg.ts:166]: The name of this method is also confusing. Remove the `className` property from here, this is only used to check properties that can be animated.

File: src/@utils/svg.ts [hunk 4]
```diff
@@ -172,7 +173,8 @@ export const getSVGProperty = (property: string): SVGNativeProperties => {
         fillOpacity: 'fill-opacity',
         strokeColor: 'stroke',
         strokeOpacity: 'stroke-opacity',
-        strokeWidth: 'stroke-width'
+        strokeWidth: 'stroke-width',
+        className: 'class'
```
Comment (elchininet) [inline src/@utils/svg.ts:177]: The same with this one, it is used only for animatable properties.

**Review by elchininet — CHANGES_REQUESTED** [review]
LGTM, just some removals are required.

File: package-lock.json (hunk 1 above)
Comment (joelburget) [reply package-lock.json]: Whoops didn't realize I added this

**Comment by elchininet** [comment]
@joelburget,
It is ready to merge, rebase with master and solve the conflicts with `dist/web/isometric.js`.

**Review by elchininet — APPROVED** [review]

**Comment by joelburget** [comment]
I just merged and rebuilt `dist/web/isometric.js` (`git p --no-rebase upstream master; npm run build; git add dist/web/isometric.js`)

**Pull request merged**